htdocs/mri/jiv/get_file.php

```diff
@@ -92,58 +132,68 @@
     exit(4);
 }
 
+$PathPrefix = '';
 switch($FileExt) {
 case 'mnc':
-    $FullPath         = $mincPath . '/' . $File;
+    $PathPrefix       = $mincPath;
     $MimeType         = "application/x-minc";
     $DownloadFilename = basename($File);
     break;
 case 'nii':
-    $FullPath         = $mincPath . '/' . $File;
+    $PathPrefix       = $mincPath;
     $MimeType         = "application/x-nifti";
     $DownloadFilename = basename($File);
     break;
 case 'nii.gz':
-    $FullPath         = $mincPath . '/' . $File;
+    $PathPrefix       = $mincPath;
     $MimeType         = "application/x-nifti-gz";
     $DownloadFilename = basename($File);
     break;
 case 'png':
-    $FullPath = $imagePath . '/' . $File;
-    $MimeType = "image/png";
+    $PathPrefix = $imagePath;
+    $MimeType   = "image/png";
     break;
 case 'jpg':
-    $FullPath = $imagePath . '/' . $File;
-    $MimeType = "image/jpeg";
+    $PathPrefix = $imagePath;
+    $MimeType   = "image/jpeg";
     break;
 case 'header':
 case 'raw_byte.gz':
     // JIVs are relative to imagePath for historical reasons
     // And they don't have a real mime type.
-    $FullPath = $imagePath . '/' . $File;
-    $MimeType = 'application/octet-stream';
+    $PathPrefix = $imagePath;
+    $MimeType   = 'application/octet-stream';
     break;
 case 'xml':
-    $FullPath         = $imagePath . '/' . $File;
+    $PathPrefix       = $imagePath;
     $MimeType         = 'application/xml';
     $DownloadFilename = basename($File);
     break;
 case 'nrrd':
-    $FullPath         = $imagePath . '/' . $File;
+    $PathPrefix       = $imagePath;
     $MimeType         = 'image/vnd.nrrd';
     $DownloadFilename = basename($File);
     break;
 default:
-    $FullPath         = $DownloadPath . '/' . $File;
+    $PathPrefix       = $DownloadPath;
     $MimeType         = 'application/octet-stream';
     $DownloadFilename = basename($File);
     break;
 }
+$FullPath = "$PathPrefix/$File";
+$TruePath = truepath($FullPath);
+// Check if the resolved 'true' path starts with the prefix.
+// If not, user supplied a relative path which is forbidden
+if (substr($TruePath, 0, strlen($PathPrefix)) !== $PathPrefix) {
+    error_log("ERROR: User supplied relative path. Potentially malicious.");
```

Comment:
You might want to include the IP address or username if you're going to provide a message like this, so it can be followed up with.